orca/2.1-a-first-look-at-a-neural-network.ipynb

```diff
@@ -0,0 +1,518 @@
+    "batch_size = 320"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Before training, we will preprocess our data by reshaping it into the shape that the network expects, and scaling it so that all values are in \n",
+    "the `[0, 1]` interval. We need to create a trainging and a testing dataset whose elements are slices of the original datasets.\n",
+    "Also, our training images for instance were stored in an array of shape `(60000, 28, 28)` of type `uint8` with \n",
+    "values in the `[0, 255]` interval previously. We need to transform it into a `float32` array of shape `(60000, 28 * 28)` with values between 0 and 1. In addition, the \"repeat\" and \"shuffle\" process will help the model avoid bias and converge faster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def preprocess(x, y):\n",
+    "    x = tf.cast(tf.reshape(x, (28, 28, 1)), dtype=tf.float32) / 255.0\n",
+    "    return x, y\n",
+    "\n",
+    "def train_data_creator(config):\n",
+    "    (train_feature, train_label), _ = mnist.load_data()\n",
+    "\n",
+    "    dataset = tf.data.Dataset.from_tensor_slices((train_feature, train_label))\n",
+    "    dataset = dataset.repeat()\n",
+    "    dataset = dataset.map(preprocess)\n",
+    "    dataset = dataset.shuffle(1000)\n",
+    "    dataset = dataset.batch(config[\"batch_size\"])\n",
+    "    return dataset\n",
+    "\n",
+    "def val_data_creator(config):\n",
+    "    _, (val_feature, val_label) = mnist.load_data()\n",
+    "\n",
+    "    dataset = tf.data.Dataset.from_tensor_slices((val_feature, val_label))\n",
+    "    dataset = dataset.repeat()\n",
+    "    dataset = dataset.map(preprocess)\n",
+    "    dataset = dataset.batch(config[\"batch_size\"])\n",
+    "    return dataset"
+   ]
+  },
+  {
+   "source": [
+    "We are now ready to train our network using Orca Estimator. Firstly, we need to initialize the Orca context by specifying local or distributed mode."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    
+   ],
+   "source": [
+    "from zoo.orca import init_orca_context, stop_orca_context\n",
+    "from zoo.orca import OrcaContext\n",
+    "\n",
+    "OrcaContext.log_output = False  # (True: display terminal's stdout and stderr in the Jupyter notebook).\n",
+    "cluster_mode = \"local\"\n",
+    "\n",
+    "if cluster_mode == \"local\":\n",
+    "    init_orca_context(cluster_mode=\"local\", cores=4)  # run in local mode\n",
+    "elif cluster_mode == \"k8s\":\n",
+    "    init_orca_context(cluster_mode=\"k8s\", num_nodes=2, cores=2)  # run on K8s cluster\n",
+    "elif cluster_mode == \"yarn\":\n",
+    "    init_orca_context(cluster_mode=\"yarn-client\", num_nodes=2, cores=2)  # run on Hadoop YARN cluster"
```

Comment:
Do we need all these branches? Or just keep `init_orca_context` with default arguments? @jason-dai